cpp/dolfinx/fem/FiniteElement.cpp

```diff
@@ -123,7 +123,12 @@ _extract_sub_element(const FiniteElement<T>& finite_element,
 template <std::floating_point T>
 FiniteElement<T>::FiniteElement(const ufcx_finite_element& e)
     : _signature(e.signature), _space_dim(e.space_dimension),
-      _value_shape(e.value_shape, e.value_shape + e.value_rank),
+      _reference_value_shape(e.reference_value_shape,
+                             e.reference_value_shape + e.reference_value_rank),
+      _value_shape_0(e.value_shape[0], e.value_shape[0] + e.value_rank),
+      _value_shape_1(e.value_shape[1], e.value_shape[1] + e.value_rank),
+      _value_shape_2(e.value_shape[2], e.value_shape[2] + e.value_rank),
+      _value_shape_3(e.value_shape[3], e.value_shape[3] + e.value_rank),
```

Comment:
I don't see why this is needed? Seems like redundant data.

Reply:
I've moved value_shape to function space so these are no longer needed